pkg/debugdetection/debugdetection.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debugdetection
+
+import (
+	"os"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func init() {
+	for _, arg := range os.Args {
+		if arg == "--debug" {
```

Comment:
Unfortunately this doesn't work as you were expecting because I can set up as follows:

- [ ] `--debug=true` -> doesn't work as expected
- [X] `--debug true`
- [x] `--debug`
- [ ] `--debug false` -> doesn't work as expected
- [ ] `--debug=false` -> doesn't work as expected